static/panes/device-view.ts

```diff
@@ -123,6 +131,106 @@ export class DeviceAsm extends MonacoPane<monaco.editor.IStandaloneCodeEditor, D
+                // opcode +
+                // ' documentation <sup><small class="fas fa-external-link-alt opens-new-window"' +
+                // ' title="Opens in a new window"></small></sup></a>.' +
+                '<br><br>If the documentation for this opcode is wrong or broken in some way, ' +
+                'please feel free to <a href="' +
+                newGitHubIssueUrl() +
+                '" target="_blank" rel="noopener noreferrer">' +
+                'open an issue on GitHub <sup><small class="fas fa-external-link-alt opens-new-window" ' +
+                'title="Opens in a new window"></small></sup></a>.'
+            );
+        }
+
+        try {
+            // if (this.compiler?.supportsAsmDocs) {
+                const asmHelp = await this.getAsmInfo(
+                    word.word,
+                    // this.compiler.instructionSet as AssemblyDocumentationInstructionSet,
+                    this.selectedDevice.split(' ')[0].toLowerCase() as AssemblyDocumentationInstructionSet,
+                );
+                if (asmHelp) {
+                    this.alertSystem.alert(opcode + ' help', asmHelp.html + appendInfo(asmHelp.url), {
+                        onClose: () => {
+                            ed.focus();
+                            ed.setPosition(pos);
+                        },
+                    });
+                } else {
+                    this.alertSystem.notify('This token was not found in the documentation. Sorry!', {
+                        group: 'notokenindocs',
+                        alertClass: 'notification-error',
+                        dismissTime: 5000,
+                    });
+                }
+            // }
+        } catch (error) {
+            this.alertSystem.notify('There was an error fetching the documentation for this opcode (' + error + ').', {
+                group: 'notokenindocs',
+                alertClass: 'notification-error',
+                dismissTime: 5000,
+            });
+        }
+    }
+    private alertSystem: Alert;
+
+    private async getAsmInfo(
+        opcode: string,
+        instructionSet: AssemblyDocumentationInstructionSet,
+    ): Promise<AssemblyInstructionInfo | undefined> {
+        const cacheName = `asm/${instructionSet}/${opcode}`;
+        const cached = OpcodeCache.get(cacheName);
+        if (cached) {
+            if (cached.found) return cached.data as AssemblyInstructionInfo;
+            throw new Error(cached.data as string);
+        }
+
+        const response = await getAssemblyDocumentation({opcode, instructionSet});
+        const body = await response.json();
+        if (response.status === 200) {
+            OpcodeCache.set(cacheName, {found: true, data: body});
+            return body;
+        } else {
+            const error = (body as any).error;
+            OpcodeCache.set(cacheName, {found: false, data: error});
+            throw new Error(error);
+        }
```

Comment:
It'd be nice if we could eliminate the code duplication here, maybe the Compiler's `getAsmInfo` could be made static? (`getAsmInfo` probably doesn't belong in the Compiler in the first place, some day we should refactor it but definitely doesn't have to be part of this pr)